src/app/child-dev-project/children/children-list/children-list.component.html

```diff
@@ -144,178 +144,37 @@ <h1>{{ listName }}</h1>
 
 <div class="mat-elevation-z2 table-list">
   <table mat-table [dataSource]="childrenDataSource" matSort style="width: 100%">
-    <ng-container matColumnDef="projectNumber">
-      <th mat-header-cell *matHeaderCellDef mat-sort-header>PN</th>
-      <td mat-cell *matCellDef="let child">{{ child.projectNumber }}</td>
-    </ng-container>
-
-    <ng-container matColumnDef="name">
-      <th mat-header-cell *matHeaderCellDef mat-sort-header>Name</th>
-      <td mat-cell *matCellDef="let child">
-        <app-child-block [entity]="child"></app-child-block>
-      </td>
-    </ng-container>
-
-    <ng-container matColumnDef="age">
-      <th mat-header-cell *matHeaderCellDef mat-sort-header>Age</th>
-      <td mat-cell *matCellDef="let child">{{ child.age }}</td>
-    </ng-container>
-
-    <ng-container matColumnDef="dateOfBirth">
-      <th mat-header-cell *matHeaderCellDef mat-sort-header>DoB</th>
-      <td mat-cell *matCellDef="let child">
-        {{ child.dateOfBirth | date: "longDate" }}
-      </td>
-    </ng-container>
-
-    <ng-container matColumnDef="gender">
-      <th mat-header-cell *matHeaderCellDef mat-sort-header>Gender</th>
-      <td mat-cell *matCellDef="let child">{{ child.gender }}</td>
-    </ng-container>
-
-    <ng-container matColumnDef="schoolClass">
-      <th mat-header-cell *matHeaderCellDef mat-sort-header>Class</th>
-      <td mat-cell *matCellDef="let child">{{ child.schoolClass }}</td>
-    </ng-container>
+    <ng-container *ngFor="let col of columns" [matColumnDef]="col.id">
+      <th mat-header-cell *matHeaderCellDef mat-sort-header>{{col.title}}</th>
+      <td *matCellDef="let child">
+
+        <div *ngIf="col.type === 'text'">
+          {{ child[col.id] }}
+        </div>
+
+        <div *ngIf="col.type === 'date'">
+          {{ child[col.id] | date: "longDate" }}
+        </div>
+
+        <app-child-block
+                *ngIf="col.type === 'child-block'"
+                [entity]="child"
+        ></app-child-block>
+
+        <app-list-relation
```

Comment:
Actually this is a school block - or a child-school-relation. Name should be more explicit